### ISSUE TITLE
Criteria for what constitutes a valid internet connection?

### ISSUE THREAD

**Issue by cr08** [TalalMash/Unofficial-Speedify-Installer-For-OpenWrt#7]
TL;DR version: I'm curious if it is known what specific criteria Speedify looks for to consider a connection valid for use?

Right now I'm working on a build running on a small Celeron based netbook and using a stock OpenWRT 23.05.2 x86_64 generic image and trying to use this unofficial installer. After a few hiccups along the way (I found that how this is set up doesn't seem to like working with `luci-nginx` and you should stick with just `luci` with uhttpd underneath. Something about the actual Speedify UI not being able to access some websocket resources and eventually complains about the service not running?) I managed to get it all largely working.

What I'm running into is that in my setup all my connectivity is going to be USB based in one shape or another. Currently two Wifi dongles, one for WAN and one for LAN. And there's also a built in USB connected LTE modem that will be added into the mix later. But with just the WAN card installed (named `wwan`, working connectivity confirmed, DHCP client, gateway metric set to 10) Speedify can briefly connect once and then drops the connection and refuses to see it as valid going forward. The logs also list it as invalid.

I don't have any logs or screenshots at the moment but I'll grab those and add them later today. Initially I'm just curious if there's any known criteria Speedify uses to consider a connection valid or invalid that can potentially point me in the right direction. I'm otherwise fairly comfortable with OWRT's interface configuration.

**Comment by cr08** [comment]
So never got around to getting the logs but seemed to resolve itself after toying with the interfaces and might have been after I added my LTE cards in and it kept the wifi client in there as well. But wouldn't mind having more info on this if there's anything known/documented.

That all said and unrelated: I decided to give up on this and went back to SmoothWAN on a Pi 4. There were too many headaches trying to manage devices and interfaces with all of them being USB based and having them all move around on there own at every boot. The biggest one was the two ethernet dongles I had set up as WAN and LAN kept swapping places. Did all the research and it was just a pain trying to set up something more static. I'm actually quite surprised and disappointed OpenWRT doesn't have this feature by default. At first glance things seem to be holding on the Pi 4 with SmoothWAN. At the very least the built in ethernet and Wifi chips likely help.
